From 71f457d9b232449cf75293e63f3886b2c18d631a Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@qt.io>
Date: Tue, 10 Nov 2020 13:40:46 +0100
Subject: [PATCH 061/171] Android: Add the required linker flags for unwinding
 on armeabi-v7a

These flags are indicated as required for armeabi-v7a here:
https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#Unwinding

Change-Id: Id2bbfc65c4ed9bda29692869a719dbab70891491
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit 85fe4c89835d862db30b43c775d8863fcf896f20)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 mkspecs/features/android/default_pre.prf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mkspecs/features/android/default_pre.prf b/mkspecs/features/android/default_pre.prf
index eaa2b0c666..9f90dcb391 100644
--- a/mkspecs/features/android/default_pre.prf
+++ b/mkspecs/features/android/default_pre.prf
@@ -36,6 +36,9 @@ QMAKE_CFLAGS += -fno-limit-debug-info
 
 QMAKE_LINK    = $$QMAKE_CXX $$QMAKE_CFLAGS
 
+equals(ANDROID_TARGET_ARCH, armeabi-v7a): \
+   QMAKE_LINK += -Wl,--exclude-libs,libunwind.a
+
 ANDROID_STDCPP_PATH = $$NDK_LLVM_PATH/sysroot/usr/lib/
 
 # -fstack-protector-strong offers good protection against stack smashing attacks.
-- 
2.31.1

